engine/src/main/scala/cromwell/engine/workflow/ShadowWorkflowActor.scala

```diff
@@ -85,62 +87,65 @@ object ShadowWorkflowActor {
+      val actor = context.actorOf(ShadowMaterializeWorkflowDescriptorActor.props())
+      actor ! ShadowMaterializeWorkflowDescriptorCommand(workflowId, workflowSources, conf)
       goto(MaterializingWorkflowDescriptorState)
     case Event(AbortWorkflowCommand, stateData) => ??? // TODO: Handle abort
   }
 
   when(MaterializingWorkflowDescriptorState) {
-    case Event(MaterializeWorkflowDescriptorSuccess(workflowDescriptor: WorkflowDescriptor), stateData) =>
-      val newBackendAssignments: Map[Call, String] = ??? // TODO: Assign a backend for every call
-      val initializerActor = context.actorOf(WorkflowInitializationActor.props(tag, newBackendAssignments), name = s"EngineWorkflowInitializationActor-$workflowId")
+    case Event(ShadowMaterializeWorkflowDescriptorSuccessResponse(workflowDescriptor), stateData) =>
+      val initializerActor = context.actorOf(WorkflowInitializationActor.props(workflowId, workflowDescriptor), name = s"WorkflowInitializationActor-$workflowId")
       initializerActor ! StartInitializationCommand
-      goto(InitializingWorkflowState) using stateData.copy(backendAssignments = newBackendAssignments)
-    case Event(MaterializeWorkflowDescriptorFailure(reason: Throwable), _) =>
-      // TODO: Handle workflow failure
-      ???
+      goto(InitializingWorkflowState) using ShadowWorkflowActorDataWithDescriptor(workflowDescriptor)
+    case Event(ShadowMaterializeWorkflowDescriptorFailureResponse(reason: Throwable), _) =>
+      context.parent ! ShadowWorkflowFailedResponse(workflowId, MaterializingWorkflowDescriptorState, Seq(reason))
+      goto(WorkflowFailedState)
     case Event(AbortWorkflowCommand, stateData) => ??? // TODO: Handle abort
   }
 
   when(InitializingWorkflowState) {
-    case Event(WorkflowInitializationSucceededResponse, stateData) =>
-      val executionActor = context.actorOf(WorkflowExecutionActor.props(tag, stateData.backendAssignments), name = s"EngineWorkflowExecutionActor-$workflowId")
+    case Event(WorkflowInitializationSucceededResponse, ShadowWorkflowActorDataWithDescriptor(workflowDescriptor)) =>
+      val executionActor = context.actorOf(WorkflowExecutionActor.props(workflowId, workflowDescriptor), name = s"WorkflowExecutionActor-$workflowId")
       val commandToSend = startMode match {
         case StartNewWorkflow => StartExecutingWorkflowCommand
         case RestartExistingWorkflow => RestartExecutingWorkflowCommand
       }
       executionActor ! commandToSend
       goto(ExecutingWorkflowState)
-    case Event(WorkflowInitializationFailedResponse(reasons), _) =>
-      // TODO: Handle workflow failure
-      ???
+    case Event(WorkflowInitializationFailedResponse(reason), _) =>
+      context.parent ! ShadowWorkflowFailedResponse(workflowId, InitializingWorkflowState, reason)
+      goto(WorkflowFailedState)
     case Event(AbortWorkflowCommand, stateData) => ??? // TODO: Handle abort
```

Comment:
I assumed that they would be different depending on the stage. TBH I wouldn't spend time worrying about these aborts since it's all going to have to change regardless